sdk/src/azure/iot/az_iot_adu_client.c

```diff
@@ -72,7 +72,10 @@
 #define AZ_IOT_ADU_CLIENT_AGENT_PROPERTY_NAME_CREATED_DATE_TIME "createdDateTime"
 
 #define NULL_TERM_CHAR_SIZE 1
-#define UPDATE_ID_ESCAPING_CHARS_LENGTH 24
+
+#define RESULT_STEP_ID_PREFIX "step_"
+#define MAX_UINT32_NUMBER_OF_DIGITS 10
+#define RESULT_STEP_ID_MAX_SIZE (sizeof(RESULT_STEP_ID_PREFIX) + MAX_UINT32_NUMBER_OF_DIGITS)
```

Comment:
1. are any of these used?
2. Suggestion:

```suggestion
#define RESULT_STEP_ID_PREFIX "step_"
#define MAX_UINT32_NUMBER_OF_DIGITS 10
#define RESULT_STEP_ID_MAX_SIZE (sizeof(RESULT_STEP_ID_PREFIX) - 1 + MAX_UINT32_NUMBER_OF_DIGITS)
```

Minus one needed?

Reply:
Yes, it's used to generate the results for the instructions steps in the Agent state.
Your suggestion for the -1 is good. Taken.